app/src/main/java/org/grapheneos/pdfviewer/PdfViewer.java

```diff
@@ -217,6 +219,31 @@ public void onPageFinished(WebView view, String url) {
             }
         });
 
+        final GestureDetector detector = new GestureDetector(PdfViewer.this,
+                new GestureDetector.SimpleOnGestureListener() {
+                    @Override
+                    public boolean onSingleTapUp(MotionEvent motionEvent) {
+                        if (mUri != null) {
+                            mWebView.evaluateJavascript("isTextSelected()", selection -> {
```

Comment:
Why does it need to check for text selection? Is it because it would lose the selection by doing this?

Reply:
This is to emulate the standard behaviour of other PDF viewers.
When text is selected, immersive mode should not be entered/exited,
the selection should be cleared beforehand.